### PR TITLE
betterlockscreen 3.2.0 -> 4.0.0

### PR DESCRIPTION
**Pull request by SebTM**
###### Motivation for this change
I've mainly joined/revived the Betterlockscreen project about three months ago. Today after a beta-phase I released the next stable major-release prepared the last months :)

###### Things done
Updated version, adjusted dependencies according the shell.nix (also provided in the project), Added as maintainer for the package and so added to maintainers 

I will also fix a back-port if the bot isn't getting it :)

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/screensavers/betterlockscreen/default.nix [hunk 1]
```diff
@@ -1,39 +1,51 @@
...
-  patches = [ ./replace-i3lock.patch ];
-
   installPhase =
     let
       PATH =
         lib.makeBinPath
-        [imagemagick i3lock-color xdpyinfo xrandr bc feh procps xrdb];
+        [ bc coreutils i3lock-color gawk gnugrep gnused imagemagick procps xdpyinfo xrandr xset ];
     in ''
       mkdir -p $out/bin
```
Comment (Artturin) [inline pkgs/misc/screensavers/betterlockscreen/default.nix:39]:
```suggestion
	  runHook preInstall
      mkdir -p $out/bin
```

File: pkgs/misc/screensavers/betterlockscreen/default.nix [hunk 2]
```diff
@@ -1,39 +1,51 @@
...
       cp betterlockscreen $out/bin/betterlockscreen
       wrapProgram "$out/bin/betterlockscreen" --prefix PATH : "$out/bin:${PATH}"
```
Comment (Artturin) [inline pkgs/misc/screensavers/betterlockscreen/default.nix:41]:
```suggestion
      wrapProgram "$out/bin/betterlockscreen" --prefix PATH : "$out/bin:${PATH}"
      runHook postInstall
```

**Review by Artturin — CHANGES_REQUESTED** [review]
uncapitalize the commit

File: pkgs/misc/screensavers/betterlockscreen/default.nix (hunk 2 above)
Comment (SebTM) [reply pkgs/misc/screensavers/betterlockscreen/default.nix:43]: Thanks for your feedback, everything resolved :)

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133306` at 2f59b54e run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>betterlockscreen</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook postInstall`.

Near pkgs/misc/screensavers/betterlockscreen/default.nix:33:3:
```
   |
33 |   installPhase =
   |   ^
```
</li>
  </ul>
</details>

**Review by Artturin — APPROVED** [review]

**Pull request merged**